tests/PatternAssertions/BasePatternAssert.php

```diff
@@ -58,6 +59,9 @@ public function assertPattern(array $expected, string $html): void {
     $assertion_map = $this->getAssertions($variant);
     $crawler = new Crawler($html);
     foreach ($expected as $name => $expected_value) {
+      if (!array_key_exists($name, $assertion_map)) {
+        throw new Exception(sprintf('The expected key %s does not correspond to any of the pattern expected keys.', $name));
+      }
       if (isset($assertion_map[$name]) && is_array($assertion_map[$name]) && is_callable($assertion_map[$name][0])) {
```

Comment:
Minor comment. I suppose `isset($assertion_map[$name]) &&` can be removed.
```suggestion
      if (is_array($assertion_map[$name]) && is_callable($assertion_map[$name][0])) {
```